### PR TITLE
Triggers

### PR DESCRIPTION
**Pull request by skalnik**
:tada:

### CODE REVIEW

File: scripts/triggers.coffee [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#   trigger remove phrase        - Delete the given trigger
+#
+# Author:
+#   skalnik
+
+
+module.exports = (robot) ->
+  robot.respond /trigger (.*) with (.*)/, (msg) ->
+    trigger = msg.match[1]
+    response = msg.match[2]
```
Comment (sshirokov) [inline scripts/triggers.coffee]: You should `trim()` both of those, for sanity's sake

File: scripts/triggers.coffee [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    oldResponse = robot.brain.get("trigger:#{trigger}")
+    if oldResponse?
+      msg.send "'#{trigger}' already triggers '#{response}'. Use trigger remove #{trigger} to delete it."
+    else
+      robot.brain.set("trigger:#{trigger}", response)
+      robot.brain.set('triggers', robot.brain.get('triggers').push(trigger)
```
Comment (sshirokov) [inline scripts/triggers.coffee]: Missing `)` at the end of line

**Comment by sshirokov** [comment]
I'm not super into the use of both `triggers` -> List and `trigger:$str` -> String

I also MAY not be a huge fan of storing the map as a direct mapping of `trigger:$str` -> String rather than something like `trigger:md5($str)` -> String, but it's almost a detail.

File: scripts/triggers.coffee [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+#   Triggers allow you to make the robot respond with the given string.
+#
+# Dependencies:
+#   None
+#
+# Configuration:
+#   None
+#
+# Commands:
+#   trigger phrase with response - Make the robot respond with "response" when it hears "phrase"
```
Comment (skalnik) [inline scripts/triggers.coffee]: This sounds weird. I'd like to improve this but I'm not sure what to replace it with :confounded:

**Comment by skalnik** [comment]
@sshirokov :eyes: :question:

File: scripts/triggers.coffee [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      delete triggers[trigger]
+      robot.brain.set('triggers', triggers)
+      msg.send "I have forgot what to say when I hear '#{trigger}'"
+    else
+      msg.send "I have no idea what to say when I hear '#{trigger}'"
+
+  robot.hear /(.*)/, (msg) ->
+    triggers = robot.brain.get('triggers')
+    fullMessage = msg.match[1].trim()
+    for trigger in Object.keys(triggers)
```
Comment (sshirokov) [inline scripts/triggers.coffee]: `for own trigger, reply of triggers` is probos a bit nicer here

**Comment by sshirokov** [comment]
> :question: 

How do you feel about tests? :trollface:

**Comment by skalnik** [comment]
@sshirokov :cat: :thought_balloon::question:

**Comment by sshirokov** [comment]
Is it worth making the triggers case insensitive, or is this Right On:tm:

File: scripts/triggers.coffee [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    trigger = msg.match[1].trim()
+
+    if triggers[trigger]?
+      delete triggers[trigger]
+      robot.brain.set('triggers', triggers)
+      msg.send "I have forgot what to say when I hear '#{trigger}'"
+    else
+      msg.send "I have no idea what to say when I hear '#{trigger}'"
+
+  robot.hear /(.*)/, (msg) ->
```
Comment (sshirokov) [inline scripts/triggers.coffee]: This triggers as soon as you add a trigger.  :cry:

Comment (skalnik) [inline scripts/triggers.coffee]: :wrench:

**Comment by skalnik** [comment]
> Is it worth making the triggers case insensitive

:+1:

File: scripts/triggers.coffee (hunk 5 above)
Comment (sshirokov) [inline scripts/triggers.coffee]: :fire:

Comment (skalnik) [inline scripts/triggers.coffee]: :fire_engine:

**Pull request merged**